### PR TITLE
Add support for argparse as either a definition or overlay value source

### PR DESCRIPTION
**Pull request by twobraids**
This is support for configman being a drop in replacement/enhancement for argparse.  It does not make argparse the default commandline processing module, it just makes argparse available.

doc changes, demo and tutorial changes will be in the next PR.

### CODE REVIEW

File: configman/__init__.py [hunk 1]
```diff
@@ -26,7 +26,7 @@
 from configman.environment import environment
 # this next line brings in command_line and, if argparse is available,
 # a definition of the configman version of ArgumentParser.  Why is it done
-# with "import *" ? Because we don't know what symbols to import, the decision
+# with "import *" ? Because I don't know what symbols to import, the decision
```
Comment (twobraids) [inline configman/__init__.py]: hmmm, not supposed to be part of this PR -- I'll remove this

File: configman/def_sources/for_argparse.py [hunk 2]
```diff
@@ -2,13 +2,675 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-# this is a stub for future implementation
+"""this module introduces support for argparse as a data definition source
+for configman.  Rather than write using configman's data definition language,
+programsd can instead use the familiar argparse method."""
```
Comment (twobraids) [inline configman/def_sources/for_argparse.py]: typo

why don't I ever see these things before I submit the PR?

File: configman/tests/test_config_manager.py [hunk 3]
```diff
@@ -13,6 +13,9 @@
 
 import mock
 
+import argparse
```
Comment (twobraids) [inline configman/tests/test_config_manager.py]: dang it, this needs to be protected in case argparse doesn't exist.

**Comment by peterbe** [comment]
My browser stutters just to load the massive diff here on Github. I don't have time to do a thorough review. Are there any specific things you want me to look at?

**Comment by twobraids** [comment]
most of the changes to existing files were in that last PR.  This PR is primarily new files that are fairly well isolated from the rest of the code.  

@peterbe the best thing for you to do is to try using this branch with whatever you have lying around that uses argparse, keeping in mind that "argument groups" are not yet supported by my argparse wrapper.   Subparsers are supported, however.  I just need some reality testing from which to develop more unit tests.  If you find things that aren't working properly, that'll give me targets for repairs.

**Comment by peterbe** [comment]
So I struggle to get around to this one. I only have one script that uses argparse and I'm not sure how to switch it to use configman instead. 

Considering that this feature is quite academic and not something we're using in socorro, can't we just merge it. After all, we looked at much of it previously already.

**Comment by twobraids** [comment]
I'll accept this and merge it.  While I don't think the feature is academic, I think it will become the primary way that configman is used.

**Pull request merged**